rpc/src/services/protocol/mod.rs

```diff
@@ -820,30 +908,24 @@ pub(crate) fn get_context_protocol_params(
         }
     };
 
-    let protocol_hash: ProtocolHash;
-    let constants: Vec<u8>;
-    {
-        let context = env.tezedge_context();
-        let context_hash = block_header.header.context();
-
-        if let Some(data) =
-            context.get_key_from_history(context_hash, context_key_owned!("protocol"))?
-        {
-            protocol_hash = ProtocolHash::try_from(data)?;
-        } else {
-            return Err(ContextParamsError::NoProtocolForBlock(
-                block_hash.to_base58_check(),
-            ));
-        }
+    let protocol_hash =
+        match BlockMetaStorage::new(env.persistent_storage()).get_additional_data(block_hash)? {
```

Comment:
@adonagy 
also here you can use cached version: `base_services::get_additional_data_or_fail`,
after bootstrapped, this call is warmed up, so this will be even much faster